docker/buildDocker.sh

```diff
@@ -76,12 +86,58 @@ jdkVersionList()
 		- jdk13u"
 }
 
+checkArgs()
+{
+	if [[ "$useEclipseDockerFiles" == true && "$useEclipseDockerSlavesFiles" == true ]]; then
+		echo "Unable to use both kinds of dockerfiles at once, Select a single option."
+		exit 1
+        fi
+}
+
 removeBuild()
 {
 	echo "Removing ../openjdk_build/workspace/build"
 	cd $WORKSPACE/DockerBuildFolder/openjdk-build/workspace && rm -r build
 }
 
+useEclipseDockerFiles()
+{
+	cd $WORKSPACE/DockerBuildFolder/openjdk-build/docker && mkdir -p EclipseDockerfiles
+	cd EclipseDockerfiles
+	for jdk in $jdkVersion
+	do
+		# ${jdk%?} will remove the 'u' from 'jdk__u' when needed.
+		curl -o Dockerfile.$jdk https://raw.githubusercontent.com/eclipse/openj9/master/buildenv/docker/${jdk%?}/x86_64/ubuntu16/Dockerfile;
+		sharedDockerCommands $jdk
+	done
+}
+
+useEclipseDockerSlavesFiles()
+{
+	cd $WORKSPACE/DockerBuildFolder/
+	git clone https://github.com/eclipse/openj9 && cd openj9/buildenv/jenkins/docker-slaves/x86/centos6.9/
+	if [ -f "known_hosts" ]; then 
+		rm known_hosts
+	fi
+	ssh-keyscan github.com >> $PWD/known_hosts
+	cp $HOME/.ssh/id_rsa.pub $PWD
+	mv id_rsa.pub authorized_keys
+	for jdk in $jdkVersion
+	do
+		cp Dockerfile $PWD/Dockerfile.$jdk
+		sharedDockerCommands $jdk
+	done
+}
+
+sharedDockerCommands()
+{
+	local jdk=$1
+	docker build -t $jdk -f Dockerfile.$jdk .
+	docker run -it -u root -d --name=$jdk $jdk
```

Comment:
Was the root user tested with the `docker-slaves` files? Generally we use the `jenkins` user so I'm not sure the env will work. For example, the freemarker.jar is in `/home/jenkins`. There is also a line about ulimits for the user. I think that's a test req. Also, if you do end up using the Jenkins user, I seem to recall there was env differences depending on if you started the container with `u jenkins` vs starting as `root` and then `su - jenkins`. Possibly related to openssl. Those issues may have been resolved though.

Reply:
Running `./buildDocker.sh -v jdk8u --use-eclipse-docker-slave-files` as is, does seem to work, giving this output:

```
/usr/bin/make -s VERBOSE="-s" LOG_LEVEL="warn" -I /root/openj9-openjdk-jdk8/make/common -f /root/openj9-openjdk-jdk8/closed/OpenJ9.gmk SPEC=/root/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/spec.gmk openj9_test_image
/root/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2re-image/bin/java -version 2>&1 | /usr/bin/tee /root/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/test/openj9/java-version.txt
openjdk version "1.8.0_232-internal"
OpenJDK Runtime Environment (build 1.8.0_232-internal-_2019_09_30_07_12-b00)
Eclipse OpenJ9 VM (build master-230f0fa, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20190930_000000 (JIT enabled, AOT enabled)
OpenJ9   - 230f0fa
OMR      - fd3d7a7
JCL      - aaea930 based on jdk8u232-b07)
----- Build times -------
Start 2019-09-30 07:12:25
End   2019-09-30 07:41:21
00:00:28 corba
00:00:33 demos
00:02:03 docs
00:01:58 images
00:00:15 jaxp
00:00:26 jaxws
00:03:57 jdk
00:00:24 langtools
00:00:11 nashorn
00:28:56 TOTAL
-------------------------
```